src/main/java/com/flightstats/hub/app/ShutdownManager.java

```diff
@@ -62,6 +63,8 @@ public boolean shutdown(boolean useLock) throws Exception {
         }
 
         HubServices.stopAll();
+        S3WriteQueue s3WriteQueue = HubProvider.getInstance(S3WriteQueue.class);
+        s3WriteQueue.close();
```

Comment:
This method was not being called before. Not sure it is needed. If we make S3WriteQueue an AbstractService as suggested below the ServiceManager class would handle this

Reply:
I wonder why it wasn't being called by accident or by design (i.e. it was called and they stopped but never removed the code). Either way, I like the idea of dealing with all this as a service.